packages/react-components/react-table/src/hooks/useColumnResizeState.ts

```diff
@@ -0,0 +1,85 @@
+import { useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import * as React from 'react';
+import {
+  TableColumnDefinition,
+  TableColumnId,
+  ColumnResizeState,
+  ColumnWidthState,
+  UseColumnSizingParams,
+} from './types';
+import {
+  columnDefinitionsToState,
+  adjustColumnWidthsToFitContainer,
+  getColumnById,
+  setColumnProperty,
+  getTotalWidth,
+  getLastColumn,
+  getColumnByIndex,
+  getColumnWidth,
+  getLength,
+} from '../utils/columnResizeUtils';
+
+export function useColumnResizeState<T>(
+  columns: TableColumnDefinition<T>[],
+  containerWidth: number,
+  params: UseColumnSizingParams = {},
+): ColumnResizeState {
+  const { onColumnResize, columnSizingOptions } = params;
+
+  const [state, setState] = React.useState<ColumnWidthState[]>(
```

Comment:
(in between an issue and a nit)

It might make more sense here to store the state in an es6 map. That should remove the need for utility functions to get columns and all the calls of `find` around the place. Maps are still iterable using an iterator